### PR TITLE
feat(cmd): add fish completion

### PR DESCRIPTION
**Pull request by chenrui333**
Add fish completion for the CLI

### CODE REVIEW

File: internal/pkg/describe/mocks/mock_status.go [hunk 1]
```diff
@@ -3,311 +3,3 @@
 
 // Package mocks is a generated GoMock package.
 package mocks
-
-import (
-	reflect "reflect"
```
Comment (chenrui333) [inline internal/pkg/describe/mocks/mock_status.go]: This is deleted by `make gen-mocks`.

File: internal/pkg/cli/completion.go [hunk 2]
```diff
@@ -67,10 +74,16 @@ The code must be evaluated to provide interactive completion of commands.`,
   Install bash completion on linux
   /code $ source <(copilot completion bash)
   /code $ copilot completion bash > copilot.sh
-  /code $ sudo mv copilot.sh /etc/bash_completion.d/copilot`,
+  /code $ sudo mv copilot.sh /etc/bash_completion.d/copilot
+
+	Install fish completion
```
Comment (efekarakus) [inline internal/pkg/cli/completion.go]: Can we update the spacing for this statement so that it aligns with the other "Install" sentences 🙏

**Review by efekarakus — COMMENTED** [review]
Yay thank you! It looks good just a small sentence fix 🙏

File: internal/pkg/cli/completion.go [hunk 3]
```diff
@@ -17,24 +17,28 @@ import (
...
 		return nil
 	}
 	if opts.Shell == "zsh" {
 		return nil
 	}
-	return errors.New("shell must be bash or zsh")
+	if opts.Shell == "fish" {
+		return nil
+	}
+	return errors.New("shell must be bash or zsh or fish")
```
Comment (Lou1415926) [inline internal/pkg/cli/completion.go]: tiny nit:
```suggestion
	return errors.New("shell must be bash, zsh or fish")
```

File: internal/pkg/cli/completion.go [hunk 4]
```diff
@@ -43,16 +47,19 @@ func (opts *completionOpts) Execute() error {
...
 
-// BuildCompletionCmd returns the command to output shell completion code for the specified shell (bash or zsh).
+// BuildCompletionCmd returns the command to output shell completion code for the specified shell (bash or zsh or fish).
 func BuildCompletionCmd(rootCmd *cobra.Command) *cobra.Command {
 	opts := &completionOpts{}
 	cmd := &cobra.Command{
 		Use:   "completion [shell]",
 		Short: "Output shell completion code.",
-		Long: `Output shell completion code for bash or zsh.
+		Long: `Output shell completion code for bash or zsh or fish.
```
Comment (Lou1415926) [inline internal/pkg/cli/completion.go]:
```suggestion
		Long: `Output shell completion code for bash, zsh or fish.
```

File: internal/pkg/cli/completion.go [hunk 5]
```diff
@@ -67,10 +74,16 @@ The code must be evaluated to provide interactive completion of commands.`,
...
+
+	Install fish completion
+  /code$ copilot completion fish | source
+
+  To load completions for each session, execute once:
+  /code$ copilot completion fish > ~/.config/fish/completions/copilot.fish`,
 		Args: func(cmd *cobra.Command, args []string) error {
 			if len(args) != 1 {
-				return errors.New("requires a single shell argument (bash or zsh)")
+				return errors.New("requires a single shell argument (bash or zsh or fish)")
```
Comment (Lou1415926) [inline internal/pkg/cli/completion.go]:
```suggestion
				return errors.New("requires a single shell argument (bash, zsh or fish)")
```

File: internal/pkg/cli/completion_test.go [hunk 6]
```diff
@@ -25,9 +25,13 @@ func TestCompletionOpts_Validate(t *testing.T) {
...
 			wantedError: nil,
 		},
+		"fish": {
+			inputShell:  "fish",
+			wantedError: nil,
+		},
 		"invalid shell": {
 			inputShell:  "chicken",
-			wantedError: errors.New("shell must be bash or zsh"),
+			wantedError: errors.New("shell must be bash or zsh or fish"),
```
Comment (Lou1415926) [inline internal/pkg/cli/completion_test.go]:
```suggestion
			wantedError: errors.New("shell must be bash, zsh or fish"),
```

File: site/content/docs/commands/completion.en.md [hunk 7]
```diff
@@ -4,7 +4,7 @@ $ copilot completion [shell] [flags]
 ```
 
 ## What does it do?
-`copilot completion` prints shell completion code for bash or zsh. The code must be evaluated to provide interactive completion of commands.
+`copilot completion` prints shell completion code for bash or zsh or fish. The code must be evaluated to provide interactive completion of commands.
```
Comment (Lou1415926) [inline site/content/docs/commands/completion.en.md]:
```suggestion
`copilot completion` prints shell completion code for bash, zsh or fish. The code must be evaluated to provide interactive completion of commands.
```

**Review by Lou1415926 — COMMENTED** [review]
This is awesome! Thank you so much for helping 🙏🏼 🥳.  

I just have one small nits about the wording, (i.e. `"bash", "zsh" or "fish"` instead of  `"bash" or "zsh" or "fish"` for smother reading experience). All of review my comments are for the same nits, just different places.

File: internal/pkg/cli/completion.go (hunk 2 above)
Comment (chenrui333) [reply internal/pkg/cli/completion.go]: Sure, totally.

**Comment by chenrui333** [comment]
Thanks for fixing the wording issues.

**Comment by chenrui333** [comment]
I think the PR is ready to be merged. 🎉

**Review by Lou1415926 — APPROVED** [review]
Awesomee!! Thank you so much for updating this!

**Review by efekarakus — APPROVED** [review]

**Pull request merged**